x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/TraceSummary.tsx

```diff
@@ -0,0 +1,255 @@
+import { ErrorCountBadge } from './ErrorCountBadge';
+import { truncate } from '../../../../style/variables';
+
+// TODO: Light/Dark theme (@see https://github.com/elastic/kibana/issues/44840)
+const theme = euiLightVars;
+
+const Heading: React.FunctionComponent = () => (
+  <EuiTitle size="xs">
+    <h5>
+      {i18n.translate('xpack.apm.transactionDetails.traceSampleTitle', {
+        defaultMessage: 'Trace sample'
+      })}
+    </h5>
+  </EuiTitle>
+);
+
+const Item = styled('span')`
+  white-space: nowrap;
+`;
+
+interface TimestampProps {
+  transactionTimestamp: string;
+}
+const Timestamp: React.FunctionComponent<TimestampProps> = ({
+  transactionTimestamp
+}) => {
+  const timestamp = moment(transactionTimestamp).format(
+    'MMMM Do YYYY, HH:mm:ss.SSS'
+  );
+
+  return (
+    <Item>
+      <EuiToolTip
+        content={i18n.translate('xpack.apm.transactionDetails.timestampLabel', {
+          defaultMessage: 'Timestamp'
+        })}
+      >
+        <>{timestamp}</>
+      </EuiToolTip>
+    </Item>
+  );
+};
+
+const SeparatorText = styled(EuiText)`
+  color: ${theme.euiColorMediumShade};
+  display: inline;
+`;
+const Separator: React.FunctionComponent = () => (
+  <>
+    {' '}
+    <SeparatorText aria-hidden={true}>|</SeparatorText>{' '}
+  </>
+);
+
+interface DurationProps {
+  duration: number;
+  totalDuration?: number;
+}
+const Duration: React.FunctionComponent<DurationProps> = ({
+  duration,
+  totalDuration
+}) => {
+  totalDuration = totalDuration || duration;
+  const isOver100 = duration > totalDuration;
+  const percentOfTrace = isOver100
+    ? '>100%'
+    : asPercent(duration, totalDuration, '');
+  const percentOfTraceText = i18n.translate(
+    'xpack.apm.transactionDetails.percentOfTrace',
+    {
+      defaultMessage: '{value} of trace',
+      values: { value: percentOfTrace }
+    }
+  );
+
+  return (
+    <Item>
+      <EuiToolTip
+        content={i18n.translate('xpack.apm.transactionDetails.durationLabel', {
+          defaultMessage: 'Duration'
+        })}
+      >
+        <>{asTime(duration)}</>
+      </EuiToolTip>{' '}
+      {isOver100 ? (
+        <EuiToolTip
+          content={i18n.translate(
+            'xpack.apm.transactionDetails.percentOfTraceLabelExplanation',
+            {
+              defaultMessage:
+                'The % of trace exceeds 100% because this transaction takes longer than the root transaction.'
+            }
+          )}
+        >
+          <>({percentOfTraceText})</>
+        </EuiToolTip>
+      ) : (
+        `(${percentOfTraceText})`
+      )}
+    </Item>
+  );
+};
+
+const MethodText = styled('span')`
+  font-weight: bold;
+`;
+
+interface HttpStatusBadgeProps {
+  status?: number;
+}
+const HttpStatusBadge: React.FunctionComponent<HttpStatusBadgeProps> = ({
+  status
+}) => {
+  let color = 'default';
+
+  if (!status) {
+    return null;
+  } else if (status < 300) {
+    color = theme.euiColorVis0;
+  } else if (status >= 300 || status < 400) {
```

Comment:
Shouldn't this be `&&` instead of `||`? afaict anything above 300 will always return `euiColorVis5`